File: AalsiTrader/Models/Agent.swift
import Foundation
import SwiftUI

struct Agent: Codable, Identifiable {
    let id: String
    var name: String
    var greek: String
    var role: String
    var color: String
    var status: AgentStatus
    var currentTask: String?
    var lastActivity: String?
    var stats: AgentStats?

    var displayColor: Color {
        AgentDefinitions.color(for: id)
    }
}

enum AgentStatus: String, Codable {
    case active
    case sleeping
    case error
}

struct AgentStats: Codable {
    var tasksCompleted: Int?
    var alertsSent: Int?
    var accuracy: Double?
}

File: AalsiTraderAndroid/app/src/main/java/com/aalsitrader/android/model/Agent.kt
package com.aalsitrader.android.model

import kotlinx.serialization.Serializable

@Serializable
data class Agent(
    val id: String,
    val name: String = "",
    val greek: String = "",
    val role: String = "",
    val color: String = "",
    val status: AgentStatus = AgentStatus.sleeping,
    val currentTask: String? = null,
    val lastActivity: String? = null,
    val stats: AgentStats? = null,
)

@Serializable
enum class AgentStatus { active, sleeping, error }

@Serializable
data class AgentStats(
    val tasksCompleted: Int? = null,
    val alertsSent: Int? = null,
    val accuracy: Double? = null,
)
